MediaGalleryUi/view/adminhtml/web/js/grid/massaction/massactionView.js

```diff
@@ -0,0 +1,160 @@
+            this.changePageTitle();
+            this.switchButtons();
+            this.handleItemsUpdates();
+        },
+
+        /**
+         * Hide or show buttons per active mode.
+         */
+        switchButtons: function () {
+
+            if (this.massActionMode()) {
+                this.activateMassactionButtonView();
+            } else {
+                this.revertButtonsToDefaultView();
+            }
+        },
+
+        /**
+         * Sets buttons to default regular -mode view.
+         */
+        revertButtonsToDefaultView: function () {
+            $(this.deleteButtonSelector).replaceWith(this.originDeleteSelector);
+
+            if (!this.isCancelButtonInserted) {
+                $(this.cancelMassactionButtonSelector).replaceWith(this.originCancelEvent);
+            } else {
+                $(this.cancelMassactionButtonSelector).addClass('no-display');
+            }
+            $(this.imageItemSelector).css('pointer-events', '');
+
+            $.each(this.buttonsIds, function (key, value) {
+                $(value).removeClass('no-display');
+            });
+
+            $(this.assSelectedButtonSelector).addClass('no-display');
+            $(this.deleteButtonSelector).addClass('no-display media-gallery-actions-buttons');
+            $(this.deleteButtonSelector).removeClass('primary');
+        },
+
+        /**
+          * Activate mass action buttons view
+          */
+        activateMassactionButtonView: function () {
+            this.originDeleteSelector = $(this.deleteButtonSelector).clone(true, true);
+            this.originCancelEvent = $(this.cancelMassactionButton).clone(true);
+
+            $.each(this.buttonsIds, function (key, value) {
+                $(value).addClass('no-display');
+            });
+
+            $(this.imageItemSelector).css('pointer-events', 'none');
+            $(this.deleteButtonSelector).removeClass('no-display media-gallery-actions-buttons');
+            $(this.deleteButtonSelector).addClass('primary');
+
+            if (!$(this.cancelMassactionButtonSelector).length) {
+                $(this.pageActionsSelector).append(this.cancelMassactionButton);
+                this.isCancelButtonInserted = true;
+            } else {
+                $(this.cancelMassactionButtonSelector).replaceWith(this.cancelMassactionButton);
+            }
+            $(this.cancelMassactionButtonSelector).on('click', function () {
+                $(window).trigger('terminateMassAction.MediaGallery');
+            });
+
+            $(this.deleteButtonSelector).off('click').on('click', function () {
+                $(this.deleteButtonSelector).trigger('massDelete');
+            }.bind(this));
+
+        },
+
+        /**
+         * Keep buttons hidden on massaction mode state when grid updated.
+         */
+        handleItemsUpdates: function () {
+            document.querySelectorAll(this.gridSelector)[0].addEventListener('DOMSubtreeModified', function () {
```

Comment:
I think it might be better to handle that by introducing `isContextMenuVisible` property to the image component and using knockout js bindings